prost-derive/src/field/scalar.rs

```diff
@@ -297,12 +296,12 @@ impl Field {
                     quote! {
                         #[doc=#get_doc]
                         pub fn #get(&self) -> #ty {
-                            ::core::convert::TryFrom::try_from(self.#ident).unwrap_or(#default)
+                            self.#ident.unwrap_or(#default)
                         }
 
                         #[doc=#set_doc]
                         pub fn #set(&mut self, value: #ty) {
-                            self.#ident = value as i32;
+                            self.#ident = value.into();
                         }
```

Comment:
Is it still beneficial to provide a `get`, `set` and `push` functions with the new wrapper? I feel like the field can be used directly with the helper functions provided by OpenEnum.

Reply:
There is some convenience in the setter method:

```rust
msg.set_kind(Kind::Foo)
```

is nicer and more type-ahead friendly than

```rust
msg.kind = Kind::Foo.into()
```

The same applies to the push method for repeated fields.

The getter hides a bit much opinionated behavior, in my opinion, so it might be better to leave it to `OpenEnum` helper methods to explicitly decide on how unknown values should be treated.

Reply:
Another thing to consider here is porting the existing code. If these methods get used instead of direct access to fields, which I assume happens a lot now because the fields are just `i32` or `Option<i32>`, updating to the version that generates `OpenEnum` would not require changes.

Reply:
I think the better API is to have some sort of `set ` operation on `OpenEnum`. Maybe something like:
```rust
msg.kind.set(Kind::Foo)
```

Well, this change will break existing code in many ways, so I think we should directly go for the best API